### PR TITLE
GET endpoints

### PR DESCRIPTION
**Pull request by Hargeon**
Add GET endpoints for fetching data from front-end

### CODE REVIEW

File: pkg/repository/request/repository.go [hunk 1]
```diff
@@ -97,3 +102,137 @@ func (repo *Repository) Update(ctx context.Context, id int64, fields map[string]
...
+			&origin.ServiceIDDB, &converted.IDDB, &converted.NameDB, &converted.SizeDB,
+			&converted.BitrateDB, &converted.ResolutionXDB, &converted.ResolutionYDB,
+			&converted.RatioXDB, &converted.RatioYDB, &converted.ServiceIDDB)
+
+		if err != nil {
+			return nil, err
+		}
+
+		// unmarshal origin video
+		repo.unmarshalDBVideo(origin)
```
Comment (gordonBusyman) [inline pkg/repository/request/repository.go:178]: There is alternative to it, called DTO (Data Transfer Object) - check and use if u like it

File: pkg/repository/request/repository.go [hunk 2]
```diff
@@ -97,3 +102,137 @@ func (repo *Repository) Update(ctx context.Context, id int64, fields map[string]
 
 	return repo.Retrieve(ctx, reqID)
 }
+
+// List returns requests
+func (repo *Repository) List(ctx context.Context, params *query.Params) ([]interface{}, error) {
```
Comment (gordonBusyman) [inline pkg/repository/request/repository.go:107]: Mate, split the file to `list.go` `update.go` `retrieve.go` and `create.go`
Do the same for the rest of entities

File: pkg/repository/request/resource.go [hunk 3]
```diff
@@ -11,21 +11,24 @@ import (
...
-	OriginalVideo  *video.Resource
-	ConvertedVideo *video.Resource
+	ID        int64 `jsonapi:"primary,requests"`
+	UserID    int64
+	VideoName string `jsonapi:"attr,video_name"`
+	Status    string `jsonapi:"attr,status,omitempty"`
+	Details   string `jsonapi:"attr,details,omitempty"`
+
+	Bitrate     int64 `jsonapi:"attr,bitrate" validate:"required_if=ResolutionX 0 ResolutionY 0 RatioX 0 RatioY 0"`
+	ResolutionX int   `jsonapi:"attr,resolution_x" validate:"required_if=Bitrate 0 RatioX 0 RatioY 0,required_with=ResolutionY"` //nolint:lll
```
Comment (gordonBusyman) [inline pkg/repository/request/resource.go:25]: What is `//nolint:lll` ??

File: pkg/repository/user/resource.go [hunk 4]
```diff
@@ -22,6 +26,6 @@ type Resource struct {
 // JSONAPILinks ...
 func (r *Resource) JSONAPILinks() *jsonapi.Links {
 	return &jsonapi.Links{
-		"self": "need add", // TODO need add link
+		"self": fmt.Sprintf("%s/api/v1/auth/me", os.Getenv("BASE_URL")),
```
Comment (gordonBusyman) [inline pkg/repository/user/resource.go:29]:
```suggestion
		"self": fmt.Sprintf("%s/api/v1/users/%v", os.Getenv("BASE_URL"), r.ID),
```

File: pkg/repository/video/repository.go [hunk 5]
```diff
@@ -65,8 +65,32 @@ func (r *Repository) Retrieve(ctx context.Context, id int64) (jsonapi.Linkable,
...
-		Scan(&video.ID, &video.Name, &video.Size, &video.Bitrate, &video.ResolutionX,
-			&video.ResolutionY, &video.RatioX, &video.RatioY, &video.ServiceID)
+		Scan(&video.ID, &video.Name, &video.Size, &video.BitrateDB, &video.ResolutionXDB,
+			&video.ResolutionYDB, &video.RatioXDB, &video.RatioYDB, &video.ServiceID)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if video.BitrateDB.Valid {
```
Comment (gordonBusyman) [inline pkg/repository/video/repository.go:75]: DTO

File: pkg/service/auth/authorization.go [hunk 6]
```diff
@@ -5,25 +5,27 @@ import (
...
 	"github.com/Hargeon/videocmprs/pkg/repository/user"
 	"github.com/Hargeon/videocmprs/pkg/service/encryption"
 	"github.com/Hargeon/videocmprs/pkg/service/jwt"
+
 	"github.com/google/jsonapi"
 )
 
-// AuthorizationService ...
-type AuthorizationService struct {
+// Service ...
```
Comment (gordonBusyman) [inline pkg/service/auth/authorization.go:17]: Comment

File: pkg/service/auth/authorization.go [hunk 7]
```diff
@@ -49,3 +51,10 @@ func (srv *AuthorizationService) GenerateToken(ctx context.Context, resource jso
 
 	return res, nil
 }
+
+// Retrieve return user params
```
Comment (gordonBusyman) [inline pkg/service/auth/authorization.go:55]:
```suggestion
// Retrieve return user resource
```

File: pkg/service/service.go [hunk 8]
```diff
@@ -15,15 +18,20 @@ type Retriever interface {
...
+type Paginator interface {
+	List(ctx context.Context, params *query.Params) ([]interface{}, error)
 }
 
-type UserService interface {
-	Creator
-	Retriever
+type Tokenable interface {
+	GenerateToken(ctx context.Context, resource jsonapi.Linkable) (jsonapi.Linkable, error)
+	Retrieve(ctx context.Context, id int64) (jsonapi.Linkable, error)
```
Comment (gordonBusyman) [inline pkg/service/service.go:27]:
```suggestion
	Retriever
```

File: pkg/service/service.go [hunk 9]
```diff
@@ -15,15 +18,20 @@ type Retriever interface {
 	Retrieve(ctx context.Context, resource jsonapi.Linkable) (jsonapi.Linkable, error)
```
Comment (gordonBusyman) [inline pkg/service/service.go:18]:
```suggestion
	Retrieve(ctx context.Context, id int64) (jsonapi.Linkable, error)
```

**Review by gordonBusyman — CHANGES_REQUESTED** [review]
1. Do something with the interfaces.. It's a mess
2. Please add `PR_TEMPLATE` and use it

File: pkg/repository/video/repository.go (hunk 5 above)
Comment (Hargeon) [reply pkg/repository/video/repository.go]: Done

File: pkg/repository/request/repository.go (hunk 1 above)
Comment (Hargeon) [reply pkg/repository/request/repository.go]: Done

File: pkg/repository/request/repository.go (hunk 2 above)
Comment (Hargeon) [reply pkg/repository/request/repository.go]: Done

File: pkg/repository/request/resource.go (hunk 3 above)
Comment (Hargeon) [reply pkg/repository/request/resource.go:25]: The line is more than 120 characters. This comment disable linter in lines

File: db/migrations/common/20211019214146_requests.sql [hunk 10]
```diff
@@ -3,6 +3,7 @@
 CREATE TABLE IF NOT EXISTS requests (
```
[the remark this answers is not in the document]
Comment (Hargeon) [reply db/migrations/common/20211019214146_requests.sql:2]: Done

File: pkg/repository/video/resource.go [hunk 11]
```diff
@@ -22,16 +22,60 @@ type Resource struct {
...
+	ResolutionY sql.NullInt32
+	RatioX      sql.NullInt32
+	RatioY      sql.NullInt32
+	ServiceID   sql.NullString
+}
+
+func (r *ResourceDTO) BuildResource() *Resource {
+	res := new(Resource)
+
+	if r.ID.Valid {
```
Comment (gordonBusyman) [inline pkg/repository/video/resource.go:42]:
```
return &Resource{
ID: dto.ID.Int64,
Name: dto.Name.String,
...
```

File: db/migrations/common/20211019214146_requests.sql [hunk 12]
```diff
@@ -3,6 +3,7 @@
...
     id BIGSERIAL NOT NULL UNIQUE PRIMARY KEY,
     user_id BIGINT REFERENCES users,
+    video_name VARCHAR(255) NOT NULL DEFAULT '',
     status VARCHAR(255) NOT NULL DEFAULT 'original_in_review',
```
[the remark this answers is not in the document]
Comment (Hargeon) [reply db/migrations/common/20211019214146_requests.sql:6]: Done

File: pkg/service/auth/authorization.go (hunk 6 above)
Comment (Hargeon) [reply pkg/service/auth/authorization.go:17]: Fixed in other pr

**Pull request merged**